.github/scripts/metrics_checker.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -uo pipefail
+
+### This script checks if any metric behavior has been modified.
+### The checks rely on the git diff against origin/main
+### It is still up to the reviewer to make sure that any tests added are needed and meaningful.
+
+# search for any "new" or modified metric emissions
+metrics_modified=$(git --no-pager diff HEAD origin/main | grep -i "SetGauge\|EmitKey\|IncrCounter\|AddSample\|MeasureSince\|UpdateFilter")
+# search for PR body or title metric references
+metrics_in_pr_body=$(echo "${PR_BODY-""}" | grep -i "metric")
```

Comment:
I think this is missing a `:` after `PR_BODY` to get the right default behavior? So:
```
metrics_in_pr_body=$(echo "${PR_BODY:-""}" | grep -i "metric")
```
Same thing below for the `PR_TITLE`.

Reply:
daniel had shared these useful examples from https://tldp.org/LDP/abs/html/parameter-substitution.html and based off of those, I figured that `PR_BODY` and `PR_TITLE` should never be null.

They get set from the github action and empty bodies or titles should come out as the empty string. If they don't (if they are set to NULL), I'd like us to know about it because something is not right.

So that's why I left out the `:` -- does that make sense?

Reply:
Yes, the logic makes sense. I think my comment wasn't clear. I was saying that without the colon it wasn't valid syntax but I was wrong. I guess the colon is optional and I've just always written it like that :) 
```
bash-3.2$ function foo { local bar; bar="${NOT_SET-default}"; echo $bar; }
bash-3.2$ foo
default
bash-3.2$ function foo { local bar; bar="${NOT_SET:-default}"; echo $bar; }
bash-3.2$ foo
default
```